src/Parser/ConstExprParser.php

```diff
@@ -47,7 +48,7 @@ public function parse(TokenIterator $tokens, bool $trimStrings = false): Ast\Con
 
 			return $this->enrichWithAttributes(
 				$tokens,
-				new Ast\ConstExpr\ConstExprFloatNode($value),
+				new Ast\ConstExpr\ConstExprFloatNode(str_replace('_', '', $value)),
```

Comment:
What is the reason for `str_replace` here? It seems there are no other text transformations done in general, for example `\d\.` is not normalized to `\d\.\d` etc.

Reply:
Keeping the underscore here would be a BC break because phpstan-src simply casts this value to int or float, and `(int) '1_000_000'` evaluates to `1`.